### ISSUE TITLE
Error: length of 'dimnames' [2] not equal to array extent

### ISSUE THREAD

**Issue by jorismq** [alexzwanenburg/familiar#65]
Thanks for your work!
I have an error when I used the package.
My data is a csv(UTF-8) made by Excel, named "mm".
There are 20 features.
But, after I ran the familiar::summon_familiar, I got an error in the "Pre-processing: Adding imputation information to features.".

Error in matrix(fit$a0[seq(lmu * nc)], nc, lmu, dimnames = list(classnames,  :  length of 'dimnames' [2] not equal to array extent

I found it very strange that if I drop 2 features("mpsG" and "ageG"), everything is just right.
"mpsG" is a factor with 3 level.("0","1","2")
"ageG" is a factor with 5 level.("1","2","3","4","5")

There is no missing value in these 2 features.

plz, help 

My code is below:
mm<- read.table('/Desktop/mm.csv',sep=",",header=TRUE,stringsAsFactors = FALSE)
mm1<-mm[,c(1:20)]
library(familiar)
library(ggplot2)
library(scales)
set.seed(18)
familiar::summon_familiar(
  data=mm1,
  experiment_dir=file.path(tempdir(), "familiar_1"),
  outcome_type="binomial",
  outcome_column="X30d",
  experimental_design="cv(bt(fs,50)+mb,2)",
  transformation_method="box_cox_winsor",
  fs_method=c("univariate_regression","lasso_binomial"),
  learner=c("glm_logistic","lasso_binomial"),
  parallel=FALSE)

**Comment by alexzwanenburg** [comment]
Thanks for reporting this issue! I am currently not sure what causes the error. I know that the error:

- is not produced within the familiar source code.
- is produced during imputation.
- is produced when at least one categorical feature is present. A dataset with all numerical features is fine.

Therefore, my current suspicion is that the error:

- is produced within glmnet package. Even so, it would be good to gracefully handle this error within familiar by identifying its root cause, and then mitigating the issue.
- very likely has to do with the presence of categorical features.
- isn't covered by any of the tests in familiar. Since imputation is tested with different constellations of categorical features, your data might represent an edge case that should be included in the tests.

I have two questions:

- Which version of familiar and glmnet are you using?
- How frequent are each of the levels of "mpsG" and "ageG"?

**Comment by jorismq** [comment]
Thanks for replying
For the questions:
1, familiar is 1.4.1    glmnet is 4.1-6

2, "mpsG"         "ageG"
0  75.4          1   30.8%
1  24.0          2   26.6%
2  0.6            3   25.8%
                     4   13.1%
                     5   3.8

**Comment by alexzwanenburg** [comment]
I was not able to reproduce the error. Could you create a reproducible example?

**Comment by jorismq** [comment]
> 

[test.CSV](https://github.com/alexzwanenburg/familiar/files/10822065/test.CSV)

Here is my code:
rm(list = ls())
cat("\014")
gc()
.rs.restartR()
mm<- read.table('/Desktop/test.csv',sep=",",header=TRUE,stringsAsFactors = FALSE)
factor_name<-c("group","v1","v3")
idx <- which(names(mm) %in% factor_name)
for(i in idx ){
  mm[,i] <-  as.factor(mm[,i])
}
library(familiar)
library(ggplot2)
library(scales)
set.seed(18)

familiar::summon_familiar(
  data=mm,
  experiment_dir=file.path(tempdir(), "familiar_1"),
  outcome_type="binomial",
  outcome_column="group",
  experimental_design="bt(fs,100)+mb",
  transformation_method = "none",
  normalisation_method = "none",
  fs_method=c("univariate_regression","lasso_binomial"),
  learner=c("glm_logistic","lasso_binomial"),
  optimisation_metric=c("accuracy","auc","brier","npv","ppv","sensitivity","specificity"),
  optimisation_function="balanced",
  parallel=FALSE)

**Comment by alexzwanenburg** [comment]
Thanks for providing an example. I was able to reproduce the error, and fix it.

This was actually a bit of a tricky error that I think has to do with the scope where R evaluates its functions, and when.

**Comment by alexzwanenburg** [comment]
Fixed in version 1.4.2. This version can be installed from GitHub: 

```
require(devtools)
devtools::install_github("https://github.com/alexzwanenburg/familiar", dependencies=TRUE)
```
